qualification/antenna_channelised_voltage/test_vlbi.py

```diff
@@ -0,0 +1,121 @@
+    a tone as the signal. The tones are placed in the centres of the
+    channel bins to eliminate spectral leakage from the results.
+    Check that the passband ripple is at most 0.1 dB and that no alias in the
+    passband exceeds -80 dB.
+
+    The test is performed for 1712 MHz input bandwidth, 107 MHz output
+    bandwidth and 64 MHz passband.
+    """
+    receiver = receive_baseline_correlation_products
+
+    # Limit number of samples to take - otherwise the test becomes too slow.
+    step = max(1, receiver.n_chans // 64)
+    # Sample a range covering twice the bandwidth, to determine response to
+    # aliases.
+    channels = np.arange(-receiver.n_chans // 2 + step // 2, receiver.n_chans * 3 // 2, step)
+    # Determine channel to read out response from i.e. into which the input tone
+    # will alias.
+    out_channels = channels % receiver.n_chans
+
+    pdf_report.step(f"Collect response to {len(channels)} frequencies.")
+    data = await sample_tone_response_hdr(
+        cbf=cbf,
+        receiver=receiver,
+        pdf_report=pdf_report,
+        amplitude=0.99,
+        rel_freqs=channels,
+    )
+
+    pdf_report.step("Analyse results.")
+
+    data /= np.max(data)  # Normalise
+    # data contains the complete spectral response to every input, but we're
+    # only interested in the response in the corresponding channel.
+    data = data[np.arange(len(channels)), out_channels]
+    # Make relative to centre frequency
+    freqs = receiver.channel_frequency(channels) - receiver.center_freq
+    # The maximum is to avoid errors when data is 0
+    with np.errstate(divide="ignore"):  # Avoid warnings when taking log of 0
+        data_db = 10 * np.log10(data)
+
+    # Turn pass_channels (which slices the full set of channels) into
+    # a mask over the channels collected.
+    pass_select = (pass_channels.start <= channels) & (channels < pass_channels.stop)
+    ripple = np.max(data_db[pass_select]) - np.min(data_db[pass_select])
+    pdf_report.detail(f"Passband ripple is {ripple:.6f} dB.")
+    with check:
+        assert ripple < 0.1
+
+    # Mask for channels that alias into the passband
+    alias_select = (pass_channels.start <= out_channels) & (out_channels < pass_channels.stop)
+    alias_select &= ~pass_select  # Ignore the passband itself
+    max_alias = np.max(data_db[alias_select])
+    pdf_report.detail(f"Maximum alias into the passband is {max_alias:.3f} dB.")
+    with check:
+        assert max_alias < -80.0
+
+    stop_bandwidth = 2 * receiver.bandwidth - pass_bandwidth
+    fig = Figure()
+    ax = fig.add_subplot()
+    ax.axvspan(-stop_bandwidth / 2e6, -pass_bandwidth / 2e6, color="red", alpha=0.2, linewidth=0)
+    ax.axvspan(pass_bandwidth / 2e6, stop_bandwidth / 2e6, color="red", alpha=0.2, linewidth=0)
+    ax.axvline(-receiver.bandwidth / 2e6, linestyle="dashed", color="black")
+    ax.axvline(receiver.bandwidth / 2e6, linestyle="dashed", color="black")
```

Comment:
I'm not 100% following why you're dividing by `2e6`. Is it shorthand notation for `/ 2 / 1e6`?

Reply:
Yes, exactly - because the axis is in MHz.